epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/model_handling.py

```diff
@@ -1548,46 +1548,44 @@ def edit_coloramps(self, config_subproc_map):
         # will be smaller than the true number of diagram. This is fine for color
         # but maybe not for something else.
         nb_diag = max(config[0] for config in config_subproc_map)
+        import math
+        ndigits = str(int(math.log10(nb_diag))+1)
+        ndigits1 = str(int(math.log10(nb_diag))+2)
         # Output which diagrams correspond ot a channel to get information for valid color
         lines = []
-        # Note: line for index 0 (no multi-channel is hardcoded in the template, so here we fill the array from index 1)
         for diag in range(1, nb_diag+1):
+            channelidf = diag
+            channelidc = channelidf - 1 # C convention 
             if diag in diag_to_iconfig:
                 iconfigf = diag_to_iconfig[diag]
-                iconfigc = iconfigf - 1 # C convention 
-                text = "     %(iconfigc)i, // channelId=%(diag)i (diagram=%(diag)i) --> iconfig=%(iconfigf)i (f77 conv) and iconfigC=%(iconfigc)i (c conv)" 
+                iconfigftxt = '%i'%iconfigf
             else:
-                iconfigc = -1
                 iconfigf = -1
-                text = "     -1, // channelId=%(diag)i (diagram=%(diag)i): Not consider as a channel of integration (presence of 4 point interaction?)" 
-            lines.append(text % {'diag': diag, 'iconfigc':  iconfigc, 'iconfigf':iconfigf})  
-
-        replace_dict['diag_to_channel'] = '\n'.join(lines)
+                iconfigftxt = '-1 (diagram with no associated iconfig for single-diagram enhancement)'
+            text = '    %(iconfigf)NNNN1i, // CHANNEL_ID=%(channelidf)-NNNNi i.e. DIAGRAM=%(diag)-NNNNi --> ICONFIG=%(iconfigftxt)s'.replace('NNNN1',ndigits1).replace('NNNN',ndigits)
```

Comment:
Here I will get rid of the NNNN and NNNN1 local variable. For me the best is to simply keep %i here, since having nice python code is more important than having a perfectly generated code (we will need to read/update the python code more often than the generated one). But I guess that we will not agree on that.

So for me the best here is to do:
```suggestion
            text = '    %(iconfigf)i, // CHANNEL_ID=%(channelidf)i i.e. DIAGRAM=%(diag)i --> ICONFIG=%(iconfigftxt)s'
```
But anticipating that you do not like it (otherwise you would have kept it). One proposal close to what you want to do is to use a single value for the indentation. 
You can also use proper python templating (using a second templating method for simplicity). 
So for example
```suggestion
            text = '    %(iconfigf){0}i, // CHANNEL_ID=%(channelidf)-{0}i i.e. DIAGRAM=%(diag)-{0}i --> ICONFIG=%(iconfigftxt)s'.format(ndigits1)
```

Which I think is quite readable in python and super close in term of way it is printed in the .h file
(obviously in that case , it would make sense to change ndigits1 in ndigits since we will not need both)

Reply:
Thanks, second solution approved - plus I removed ndigits1 (kept only ndigits), and I removed NNNN elsewhere in the python